File: backend/utils/validators.py
import re

def validate_email(email: str) -> str | None:
    if not email or not email.strip():
        return "Email is required"
    pattern = r'^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$'
    if not re.match(pattern, email.strip()):
        return "Invalid email format"
    return None

def validate_password(password: str) -> str | None:
    if not password:
        return "Password is required"
    if len(password) < 8:
        return "Password must be at least 8 characters"
    return None

def validate_prompt(prompt: str) -> str | None:
    if not prompt or not prompt.strip():
        return "Prompt is required"
    if len(prompt.strip()) < 3:
        return "Prompt must be at least 3 characters"
    if len(prompt) > 5000:
        return "Prompt must be less than 5000 characters"
    return None

def sanitize_html(text: str) -> str:
    if not text:
        return ""
    text = text.replace("&", "&amp;")
    text = text.replace("<", "&lt;")
    text = text.replace(">", "&gt;")
    text = text.replace('"', "&quot;")
    text = text.replace("'", "&#x27;")
    return text

def validate_content_type(content_type: str) -> str | None:
    valid_types = {"blog", "social", "email", "image", "voice", "video", "bundle"}
    if content_type not in valid_types:
        return f"Invalid content type. Must be one of: {', '.join(sorted(valid_types))}"
    return None

def validate_username(username: str) -> str | None:
    if not username or not username.strip():
        return "Username is required"
    if len(username.strip()) < 3:
        return "Username must be at least 3 characters"
    if len(username) > 50:
        return "Username must be less than 50 characters"
    if not re.match(r'^[a-zA-Z0-9_]+$', username):
        return "Username can only contain letters, numbers, and underscores"
    return None

File: frontend/src/utils/validators.js
export function validatePrompt(prompt) {
  if (!prompt || prompt.trim().length === 0) {
    return 'Prompt is required';
  }
  if (prompt.trim().length < 3) {
    return 'Prompt must be at least 3 characters';
  }
  if (prompt.length > 5000) {
    return 'Prompt must be less than 5000 characters';
  }
  return null;
}

export function validateEmail(email) {
  if (!email || email.trim().length === 0) {
    return 'Email is required';
  }
  const pattern = /^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$/;
  if (!pattern.test(email.trim())) {
    return 'Invalid email format';
  }
  return null;
}

export function validatePassword(password) {
  if (!password) {
    return 'Password is required';
  }
  if (password.length < 8) {
    return 'Password must be at least 8 characters';
  }
  return null;
}

export function validateUsername(username) {
  if (!username || username.trim().length === 0) {
    return 'Username is required';
  }
  if (username.trim().length < 3) {
    return 'Username must be at least 3 characters';
  }
  if (username.length > 50) {
    return 'Username must be less than 50 characters';
  }
  if (!/^[a-zA-Z0-9_]+$/.test(username)) {
    return 'Username can only contain letters, numbers, and underscores';
  }
  return null;
}

export function sanitizeInput(text) {
  if (!text) return '';
  return text
    .replace(/&/g, '&amp;')
    .replace(/</g, '&lt;')
    .replace(/>/g, '&gt;')
    .replace(/"/g, '&quot;')
    .replace(/'/g, '&#x27;');
}
